posts/2022-04-15-libertydevops-gitjenkinshelmiks.adoc

```diff
@@ -0,0 +1,180 @@
+
+    FROM icr.io/appcafe/open-liberty:full-java11-openj9-ubi
+    # Add Liberty server configuration including all necessary features
+    COPY --chown=1001:0  server.xml /config/
+    # Modify feature repository (optional)
+    COPY --chown=1001:0 featureUtility.properties /opt/ol/wlp/etc/
+    # This script will add the requested XML snippets to enable Liberty features and grow image to be fit-for-purpose using featureUtility. 
+    # Only available in 'kernel-slim'. The 'full' tag already includes all features for convenience.
+    RUN features.sh
+    # Add interim fixes (optional)
+    #COPY --chown=1001:0  interim-fixes /opt/ol/fixes/
+    .
+    .
+    .
+    .
+    COPY --chown=1001:0 src/main/liberty/config/ /config/
+    COPY --chown=1001:0 ./target/*.war /config/apps/
+    RUN configure.sh
+
+Below given sample pipeline code can perform the code build, packaging and generating docker image and pushing the same to remote docker hub. You can use it by modifying the parameter section (<>)
+
+ pipeline {
+     agent any
+      stages {
+       stage('Build') {
+                    steps {
+              checkout([$class: 'GitSCM', branches: [[name: '*/main']], extensions: [], userRemoteConfigs: [[credentialsId: ‘<git token>, url: 'https://github.com/liberty/app.git']]])
+                    sh '''
+                         mvn -U package
+                         docker login <remote-docker-image-repository-url> -u "${USERNAME}" -p “${PASSWORD}”
+                         docker build -t liberty-$<code identifier>:$<docker image version> .
+                         #eg: docker build -t liberty-app:v1.0 .
+                         docker tag liberty-$<code identifier>:$<docker image version> <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker tag liberty-app:v1.0 docker.io/someid/liberty-app:v1.0
+                         docker push <remote-docker-image-repository-url>/<docker-repo-name>/liberty-$<code identifier>:$<docker image version>
+                         #eg: docker push docker.io/someid/liberty-app:v1.0  
+                       '''   
+                  }
+               }
+            }
+          }
+
+
+Following are the parameter that is used in this example code
+
+* Git token: Personal access token generated from github.
+* remote-docker-image-repository-url : Docker hub repository URL.
+* USERNAME/PASSWWORD: username and password to connect to docker registry.
+
+image::/img/blog/jenkins-cred-binding-and-corresponding-param.png[Pipeline credential binding and corresponding param,width=30%,align="center"]
+
+* code identifier: This is optional, a unique docker image identifier
+* docker image version: docker image version number, a unique identifier   
+
+*Security scan* can be done along with Maven build command (CLI commands way of application scanning) or can do it in a separate pipeline stage. 
```

Comment:
Rephrase to: "You can add build actions to perform static analysis of your code and security scans. For example, you can use link:[https://www.sonarqube.org/downloads/[SonarQube](https://www.sonarqube.org/downloads/%5BSonarQube) community edition] to do static code analysis, and link:[https://hub.docker.com/r/aquasec/trivy/[trivy]](https://hub.docker.com/r/aquasec/trivy/%5Btrivy%5D) to do a security scan of your built Docker image."

Reply:
This is not a correct way of explaining security scan setup and actual scan. For sonarqube scan, we need to setup scan server separately and then the scan client commands  which connects with the server for performing scan analysis. If I replace the whole explanation about scan setup as you suggested, I worry that users will be confused about the whole setup. We can discuss this further